src/main/java/com/fasterxml/jackson/core/JsonGenerator.java

```diff
@@ -1978,12 +1979,13 @@ public WritableTypeId writeTypePrefix(WritableTypeId typeIdDef) throws IOExcepti
         } else {
             // No native type id; write wrappers
             // Normally we only support String type ids (non-String reserved for native type ids)
-            String idStr = (id instanceof String) ? (String) id : String.valueOf(id);
+            String idStr = (id instanceof String) ? (String) id : Objects.toString(id, null);
```

Comment:
Nice! TIL there's 2-arg variant. Will need to start using more widely.

Reply:
I also learned it not that long time ago 😄